### PR TITLE
[FSDP] Support lr scheduler

### PR DESCRIPTION
**Pull request by ChangyiYang**
### [FSDP] Support LR scheduler

Implement `FSDPLRScheduler` for FSDP backend to align with Megatron-LM's learning rate scheduling behavior.

### Key Changes
- Add `FSDPLRScheduler` class inheriting from PyTorch `LRScheduler`
- Support warmup, decay, and weight decay scheduling (linear, cosine, etc.)
- Update actor logging and checkpoint functionality

Related issue https://github.com/THUDM/slime/issues/962

### Reference
Based on [Megatron-LM OptimizerParamScheduler](https://github.com/NVIDIA/Megatron-LM/blob/2c06b0457e4ece41fe535da49eb58b33fca54f2c/megatron/core/optimizer_param_scheduler.py)

This ensures consistent LR scheduling behavior between FSDP and Megatron backends.

### Example
image

Load and Save Test
image

### CODE REVIEW

**Comment by ChangyiYang** [comment]
@PopSoda2002  Hi! I think this pr is ready for review

File: slime/backends/fsdp_utils/lr_scheduler.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    def get_lr(self) -> list[float]:
+        """Compute the learning rates for each parameter group.
+
+        Returns:
+            list[float]: A list of learning rates, one for each parameter group.
+        """
+        return [self._get_lr_for_group(group) for group in self.optimizer.param_groups]
+
+    @override
+    def state_dict(self) -> dict:
```
Comment (Hecate0821) [inline slime/backends/fsdp_utils/lr_scheduler.py]: Is it possible to simply use ` return super().state_dict()` and `super().load_state_dict(state_dict)`?

Comment (ChangyiYang) [reply slime/backends/fsdp_utils/lr_scheduler.py]: Good point, will do

**Comment by Hecate0821** [comment]
LGTM
